### PR TITLE
docs(limits): add measurable application requirements

### PR DESCRIPTION
**Pull request by nabetetsu**
## Description

This PR will add the following conditions as measurable targets by CARET.
1. The rclcpp library must be used.
2. The processes that act as ROS node.

## Pre-review checklist for the PR author

- [x] I've confirmed the [contribution guidelines](https://github.com/tier4/caret/blob/main/.github/CONTRIBUTING.md).
- [x] The PR is ready for review.

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR has been reviewed.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

### CODE REVIEW

File: docs/design/limits_and_constraints/index.md [hunk 1]
```diff
@@ -8,7 +8,9 @@ CARET only supports environments as follows.
 - Support FastDDS or CycloneDDS
 - Support only Linux, especially Ubuntu
 - Support Galactic and Humble
-- Require rebuilding of an application
+- Support only applications build with rclcpp library
```
Comment (takeshi-iwanari) [inline docs/design/limits_and_constraints/index.md:11]:
```suggestion
- Support only applications built with rclcpp library
```

File: docs/design/limits_and_constraints/index.md [hunk 2]
```diff
@@ -8,7 +8,9 @@ CARET only supports environments as follows.
...
+- Support only processes acting as node
```
Comment (takeshi-iwanari) [inline docs/design/limits_and_constraints/index.md:12]: I don't think it is "Environment". maybe you can move this to "ROS 2 functions" or "Implementation", and description will be like "Other than processing that operates on a node. e.g. Launch"

**Review by takeshi-iwanari — APPROVED** [review]

File: docs/design/limits_and_constraints/index.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (miyakoshi-dev) [reply docs/design/limits_and_constraints/index.md:11]: I don't think it's something to write in CARET_doc, so please disregard my comment.

File: docs/design/limits_and_constraints/index.md (hunk 2 above)
Comment (nabetetsu) [reply docs/design/limits_and_constraints/index.md]: Thank you for your comment. I have fixed conditon type on [5373f84](https://github.com/tier4/CARET_doc/pull/202/commits/5373f8491a2430ab89a257d2500317c8a35008a2).

File: docs/design/limits_and_constraints/index.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (nabetetsu) [reply docs/design/limits_and_constraints/index.md]: Thank you for your comment.
I was quite unsure about this, but considering the possibility that the exclusion conditions may be changed in the future, I will not explicitly describe it in this document.

**Review by takeshi-iwanari — APPROVED** [review]
LGTM

**Review by miyakoshi-dev — APPROVED** [review]

**Comment by miyakoshi-dev** [comment]
LGTM

**Pull request merged**